tests/stress/consume_flow_with_threads_test.py

```diff
@@ -0,0 +1,164 @@
+#
+# Copyright 2022 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import os
+import random
+import shutil
+from concurrent.futures import ThreadPoolExecutor
+
+import pytest
+
+from ocean_lib.agreements.service_types import ServiceTypes
+from ocean_lib.config import Config
+from ocean_lib.data_provider.data_service_provider import DataServiceProvider
+from ocean_lib.example_config import ExampleConfig
+from ocean_lib.ocean.mint_fake_ocean import mint_fake_OCEAN
+from ocean_lib.ocean.ocean import Ocean
+from ocean_lib.structures.file_objects import FilesTypeFactory
+from ocean_lib.web3_internal.constants import ZERO_ADDRESS
+from tests.resources.ddo_helpers import (
+    build_credentials_dict,
+    get_first_service_by_type,
+)
+from tests.resources.helper_functions import (
+    deploy_erc721_erc20,
+    get_address_of_type,
+    generate_wallet,
+)
+
+
+def consume_flow(ocean: Ocean, config: Config):
+    wallet = generate_wallet()
+    metadata = {
+        "created": "2020-11-15T12:27:48Z",
+        "updated": "2021-05-17T21:58:02Z",
+        "description": "Sample description",
+        "name": "Sample asset",
+        "type": "dataset",
+        "author": "OPF",
+        "license": "https://market.oceanprotocol.com/terms",
+    }
+    data_provider = DataServiceProvider
+    file1_url = "https://raw.githubusercontent.com/tbertinmahieux/MSongsDB/master/Tasks_Demos/CoverSongs/shs_dataset_test.txt"
+    file1_dict = {"type": "url", "url": file1_url, "method": "GET"}
+    file1 = FilesTypeFactory(file1_dict)
+    file2_url = "https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-abstract10.xml.gz-rss.xml"
+    file2_dict = {"type": "url", "url": file2_url, "method": "GET"}
+    file2 = FilesTypeFactory(file2_dict)
+    file3_url = (
+        "https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-abstract10.xml.gz"
+    )
+    file3_dict = {"type": "url", "url": file3_url, "method": "GET"}
+    file3 = FilesTypeFactory(file3_dict)
+    files = [file1, file2, file3]
```

Comment:
can we maybe use some fixtures for files? It seems we are repeating this code a lot. Or maybe a function?